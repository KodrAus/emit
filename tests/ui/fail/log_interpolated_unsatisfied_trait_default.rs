#![feature(stmt_expr_attributes, proc_macro_hygiene)]

// Does not implement `Display`
struct Input;

fn main() {
    emit::info!("Text \"and\" {a: Input} and more");
}
