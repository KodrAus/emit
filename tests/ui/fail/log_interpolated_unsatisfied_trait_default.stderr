error[E0277]: `Input` doesn't implement `std::fmt::Display`
 --> fail/log_interpolated_unsatisfied_trait_default.rs:7:32
  |
7 |     emit::info!("Text \"and\" {a: Input} and more");
  |                                ^^^^^^^^ `Input` cannot be formatted with the default formatter
  |
  = help: the trait `std::fmt::Display` is not implemented for `Input`
  = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
  = help: the following other types implement trait `main::emit::emit_rt::capture::Capture<T>`:
            <(dyn std::error::Error + 'static) as main::emit::emit_rt::capture::Capture<main::emit::emit_rt::capture::CaptureError>>
            <(dyn std::fmt::Debug + 'static) as main::emit::emit_rt::capture::Capture<main::emit::emit_rt::capture::CaptureDebug>>
            <(dyn std::fmt::Display + 'static) as main::emit::emit_rt::capture::Capture<main::emit::emit_rt::capture::CaptureDisplay>>
            <str as main::emit::emit_rt::capture::Capture<T>>
  = note: required for `Input` to implement `main::emit::emit_rt::capture::Capture<main::emit::emit_rt::capture::CaptureDisplay>`
note: required by a bound in `__private_capture_as_default`
 --> $WORKSPACE/rt/src/capture.rs
  |
  |         Self: Capture<WithDefault>,
  |               ^^^^^^^^^^^^^^^^^^^^ required by this bound in `__PrivateCapture::__private_capture_as_default`
  = note: this error originates in the macro `emit::ct::__private_capture` (in Nightly builds, run with -Z macro-backtrace for more info)
